### PR TITLE
Feature/issue 783 unlimited attachment size

### LINKED ISSUE #783: allow opening messages of unlimited size + add download progress
Currently, messages with attachments over 5mb are disabled.

I'd like to re-enable them but during loading, show progress spinner. And allow user to use the back icon in case they don't want to wait.

_Sometime in the future, we'll separate the logic to only pull message body and not attachments first, but that's not trivial for OpenPGP messages (content is often in attachments) so that will later need a way to differentiate which attachments to pull. I'll file another issue for that._

### PR DESCRIPTION
**Pull request by sosnovsky**
This PR allows loading of attachments over 5mb and adds progress spinner for download process.

close #783 

----------------------------------

**Tests** _(delete all except exactly one)_:
- Difficult to test (explain why) - it's mostly UI change

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: FlowCrypt/Functionality/Mail Provider/Message Provider/Gmail+Message.swift [hunk 1]
```diff
@@ -8,34 +8,83 @@
...
-                guard let data = GTLRDecodeWebSafeBase64(raw) else {
-                    return reject(GmailServiceError.missedMessageInfo("data"))
+                guard let sizeEstimate = gmailMessage.sizeEstimate?.floatValue else {
+                    continuation.resume(throwing: GmailServiceError.missedMessageInfo("sizeEstimate"))
+                    return
                 }
-                resolve(data)
+
+                // google returns smaller estimated size
+                let totalSize = sizeEstimate * Float(1.3)
```
Comment (tomholub) [inline FlowCrypt/Functionality/Mail Provider/Message Provider/Gmail+Message.swift:73]: I did the exact same fix on browser extension. Did you get this from there? :-)

**Review by tomholub — COMMENTED** [review]
code looks good so far

File: FlowCrypt/Functionality/Mail Provider/Message Provider/Gmail+Message.swift (hunk 1 above)
Comment (sosnovsky) [reply FlowCrypt/Functionality/Mail Provider/Message Provider/Gmail+Message.swift:73]: Haha, no, I haven't seen your fix, just calculated difference between real and estimated sizes 🙂 Usually it's around 30%

**Comment by sosnovsky** [comment]
It'll be nice to have decrypt progress percentage in the future too, as it takes almost the same time as download for big files.

**Comment by sosnovsky** [comment]
@tomholub currently I have only https://github.com/FlowCrypt/flowcrypt-ios/issues/812 task assigned to me, can you please check on what tasks should I work after this one?

File: FlowCrypt/Functionality/Mail Provider/Message Provider/Gmail+Message.swift (hunk 1 above)
Comment (tomholub) [reply FlowCrypt/Functionality/Mail Provider/Message Provider/Gmail+Message.swift:73]: Careful to cap it at 100% when calculating. Else sometimes user will see 103%..

I think the difference is in base64 encoding which tends to add about a third.

Comment (sosnovsky) [reply FlowCrypt/Functionality/Mail Provider/Message Provider/Gmail+Message.swift:73]: Yes, I handle this case with setting `1` as max progress value:
```
let progress = min(Float(received)/messageSize, 1)
```

**Comment by tomholub** [comment]
> It'll be nice to have decrypt progress percentage in the future too, as it takes almost the same time as download for big files.

Download is network bound and decryption is cpu bound. You could do both at the same time, if you do streaming decryption. #230 But it's not simple to do across the Swift/JS boundary. But it's possible.

**Comment by tomholub** [comment]
> @tomholub currently I have only #812 task assigned to me, can you please check on what tasks should I work after this one?

will do

**Comment by sosnovsky** [comment]
> > It'll be nice to have decrypt progress percentage in the future too, as it takes almost the same time as download for big files.
> 
> Download is network bound and decryption is cpu bound. You could do both at the same time, and it should take half the time overall, if you do streaming decryption. #230 But it's not simple to do across the Swift/JS boundary. But it's possible.

Got it, it'll be a great addition for app's functionality and will make large emails loading much faster.

**Comment by tomholub** [comment]
I'd like you to debug/finalize #854 . How about merging sequence? I suppose it may be easier to merge the big one first and then rebase this onto that, but we could do it the other way around if you think it's easier. Up to you.

**Comment by sosnovsky** [comment]
I agree - it should be easier to merge #854 to `master` first, and then rebase it into this PR, as there is less changes here. I'll check #854 now

**Comment by tomholub** [comment]
Tested, looks great

**Comment by tomholub** [comment]
@sosnovsky good to merge once conflicts addressed 👍

**Comment by sosnovsky** [comment]
@tomholub I merged `master` changes to this branch - the only issue was with `fetchMsg` function, it didn't execute async code for fetching mail size. I fixed it by using `async` instead of `Promise`.
This fix also affected `Imap` class (to make it conform to updated `MessageProvider` protocol) - I updated `fetchMsg` function there too  https://github.com/FlowCrypt/flowcrypt-ios/pull/893/commits/763ac4b525d7cdecf34a7d971513fb8c87a26445#diff-8012f2f1a5073946e65bdfc88fd17e2b9ddad872c955b7cf6adfd7a1621bf345R12

File: FlowCrypt/Controllers/Msg/MessageViewController.swift [hunk 2]
```diff
@@ -148,26 +148,38 @@ final class MessageViewController: TableNodeViewController {
...
-        }
-        .then(on: .main) { [weak self] in
-            self?.handleReceivedMessage()
+        Task {
+            do {
+                processedMessage = try await serviceActor.fetchDecryptAndRenderMsg(message: input.objMessage,
+                                                                                   path: input.path,
+                                                                                   progressHandler: { [weak self] state in
+                                                                                        self?.handleFetchProgress(state: state)
+                                                                                   })
```
Comment (tomholub) [inline FlowCrypt/Controllers/Msg/MessageViewController.swift:161]: A detail - I'd prefer formatting like:
```swift
processedMessage = try await serviceActor.fetchDecryptAndRenderMsg(
    message: input.objMessage,
    path: input.path,
    progressHandler: { [weak self] state in
        self?.handleFetchProgress(state: state)
    }
)
```
Else all of the inputs have no space due to large indentation.

File: FlowCrypt/Controllers/Msg/MessageViewController.swift [hunk 3]
```diff
@@ -556,16 +554,20 @@ private actor ServiceActor {
...
 
     func checkAndPotentiallySaveEnteredPassPhrase(_ passPhrase: String) async throws -> Bool {
-        return try await messageService.checkAndPotentiallySaveEnteredPassPhrase(passPhrase)
+        try await messageService.checkAndPotentiallySaveEnteredPassPhrase(passPhrase)
     }
 
     func decryptAndProcessMessage(mime: Data) async throws -> ProcessedMessage {
-        return try await messageService.decryptAndProcessMessage(mime: mime)
+        try await messageService.decryptAndProcessMessage(mime: mime)
     }
```
Comment (tomholub) [inline FlowCrypt/Controllers/Msg/MessageViewController.swift:572]: I like explicit returns for the clarity :-) but don't insist.

**Review by tomholub — APPROVED** [review]
👍

File: FlowCrypt/Controllers/Msg/MessageViewController.swift (hunk 2 above)
Comment (sosnovsky) [reply FlowCrypt/Controllers/Msg/MessageViewController.swift:161]: Okay, I'll use such formatting in the next commits

**Pull request merged**